openmls/src/treesync/tests_and_kats/kats/kat_treekem.rs

```diff
@@ -0,0 +1,388 @@
+                ciphersuite.signature_algorithm(),
+                skp.signature_keypair.private().to_vec(),
+                skp.signature_keypair.to_public_vec(),
+            );
+
+            // XXX: Update own leaf.
+            let (vec_plain_update_path_nodes, _, commit_secret) = diff_after_kat
+                .apply_own_update_path(
+                    backend,
+                    &signer,
+                    ciphersuite,
+                    group_context.group_id().clone(),
+                    LeafNodeIndex::new(path_test.sender),
+                )
+                .unwrap();
+
+            // XXX: Update GroupContext.
+            let serialized_group_context = group_context.tls_serialize_detached().unwrap();
+
+            // Encrypt path to according recipients.
+            let encrypted_path = diff_after_kat
+                .encrypt_path(
+                    backend,
+                    ciphersuite,
+                    &vec_plain_update_path_nodes,
+                    &serialized_group_context,
+                    &HashSet::new(),
+                    LeafNodeIndex::new(path_test.sender),
+                )
+                .unwrap();
+
+            (
+                UpdatePath::new(
+                    diff_after_kat
+                        .leaf(LeafNodeIndex::new(path_test.sender))
+                        .unwrap()
+                        .leaf_node
+                        .clone(),
+                    encrypted_path,
+                ),
+                commit_secret,
+            )
+        };
+
+        // For each leaf node index j != i for which the leaf node is not blank:
+        //
+        //     Process new_update_path using private_leaf[j]
+        //     Verify that the resulting commit secret is new_commit_secret
+        for leaf_i in full_leaf_nodes.iter() {
+            log::trace!("   Processing self-update for leaf {}.", leaf_i.index.u32());
+
+            if leaf_i.index.u32() == path_test.sender {
+                continue;
+            }
+
+            let params = DecryptPathParams {
+                version: ProtocolVersion::Mls10,
+                update_path: update_path.nodes(),
+                sender_leaf_index: LeafNodeIndex::new(path_test.sender),
+                exclusion_list: &HashSet::default(),
+                group_context: &group_context.tls_serialize_detached().unwrap(),
+            };
+
+            let (_encryption_keys, commit_secret_inner) = tree_after_kat
+                .empty_diff()
+                .decrypt_path(
+                    backend,
+                    ciphersuite,
+                    params,
+                    &leaf_i.encryption_keys.iter().collect::<Vec<_>>(),
+                    leaf_i.index,
+                )
+                .unwrap();
+
+            log::trace!("       Successfully decrypted path secrets.");
+
+            assert_eq!(new_commit_secret, commit_secret_inner);
+        }
+    }
+}
+
+fn apply_update_path<'a>(
```

Comment:
So this is used only once now? It could be inlined then as well. But feel free to keep it here.